### DISCUSSION TITLE
opengothic-v1.0.3010

### DISCUSSION THREAD

**Discussion by Try** [Try/OpenGothic#706] [General]
<img width="1440" alt="Снимок экрана 2024-12-21 в 13 48 06" src="https://github.com/user-attachments/assets/2688d3b0-30b8-4e05-b7c3-e286e97415e7" />

| | |
|---|-------|
| ![изображение](https://github.com/user-attachments/assets/bd788005-a76d-4563-a993-85055c306512) | ![изображение](https://github.com/user-attachments/assets/52acfe69-270d-4af6-8b60-47f59fc31de9) |  
| ![изображение](https://github.com/user-attachments/assets/7474ee04-867f-4ee9-a42c-98e03ca691ff)  | ![изображение](https://github.com/user-attachments/assets/9cf18fbb-e021-4c0d-8973-2389443f69f0) |
| ![изображение](https://github.com/user-attachments/assets/6c52d12c-0c8e-4dc5-a483-e2124a204c6c) | ![изображение](https://github.com/user-attachments/assets/34ae62f3-96e4-4774-8dab-6a951dab2c0c) |

### Game
  * Improve handling of BS_STUMBLE/BS_PARADE
  * Disallow to open inventory menu, while character is laying on the ground
  * Disallow sliding, when character is in 'walk' mode
  * Perception range for npc's are now driven by game-script
  * Fixup for `isNoisy` check, when player interacting with MOBSI
  * Gothic-1: added PERC_ASSESSREMOVEWEAPON
  * Gothic-1: multiple improvements to spell-casting and alignment (#676)
  * Now, best weapons are automatically equipped by npc on spawn (alignment to original game) (#653)
  * Limit mouse rotation speed
  * Filter out mouse-wheel event with zero delta
  * Fix critical damage calculation for monsters (#599)
  * Make sure that text messages do not overlap with health-bar
  * Now npc view direction is aligned with corresponding way-point (#644)
  * Multiple improvements to friendly-fire game mechanic, for companions and enemy
  * Falling mechanic and animations are now closer to vanilla in behavior

### General
  * Marvin: add `ztimer multiplyer`
  * Marvin: "set time" minutes argument can be omitted
  * Marvin: add `goto vob` and `goto camera` commands (#634)
  * Introduce support for iOS builds
  * Log detail info for DeviceLostException
  * Added extended gpu-crash logging for Metal on MacOS and iOS
  * Use key down and support key repeat in game menu (#651)
  * Systempack compatibility: added configurable camera-FOV
  * Systempack compatibility: UI scale
  * Systempack compatibility: INTERFACE/InventoryCellSize
  * Systempack compatibility: health, mana, swim bar options
  * Systempack compatibility: added INTERFACE/SaveGameImageSizeXY
  * Added check for `useQuickSaveKeys` option when handling F5/F9 keys
  * Game-config cleanup
  * Disallow quick load while at pause
  * Improved world load time
  * Improved save-game time
  * Now 'free' camera can be moved or rotated, while game is in suspended mode
  * Hp bar for dead npc is  now hidden

### Graphics
  * Experimental support for virtual shadowmap (`-vsm 1` in command-line or `toggle vsm` in marvin)
  * Optimize shadows rendering for alpha-tested objects
  * GPU-driven rendering optimizations (#576)
  * Implement physically accurate sun color calculation, based on atmosphere properties
  * Adjust sky and fog rendering to path-traced reference
  * Fix appearance of clouds in water reflections
  * CMAA2 integration (#655)
  * Added basic(lanczos) upscaler
  * Added optimization of ray-traversal flags for AMD
  * Quality of SSAO significantly improved
  * Skip mip-map generation for texture, if original texture has no mip's
  * Added support for A4R4G4B4 texture format
  * Introduce single pass `hiz` generation

### Music
  * Introduced `dmusic` library as an option for music backend (#613)

### Script-api
  * Added wld_removenpc
  * Initial support for cutscene-camera(CsCamera, #531)
  * npc_changeattribute: don't reduce hitpoints of immortal npcs (#631)
  * npc_changeattribute: don't reduce hitpoints in godmode (#586)
  * wld_getmobstate: function now detects nearest mob (#629). This change is needed for correct work of some Gothic1 scripts
  
### Bug fixes
  * Fix some Gothic 1 magic bugs (#682, #685)
  * Fix npc-to-ray intersection test
  * Fix possible use-after-free error in key-dispather
  * Fix ini parsing: handle multiple values in the one line (#672)
  * Fix(perf) workgroup size in copy-image
  * Allow exit from load menu after trying to load empty save (#650)
  * Do not apply TA(daily routines) to npc controlled by player
  * Don't use a time limit for non-daily routine states (#648)
  * Drop items from right hand instead of left hand (#654)
  * Do not crash, if font-file is missing
  * Allow spaces, tabs and unknown words in GAME/invCatOrder
  * Added workaround for AMD driver issue with mesh shader (#583)
  * Fix handling of Metal shading language versions
  * Fix Gothic 1 melee blocking (#630)
  * Fix spell charging animation (#659)
  * Fix weapon undraw sound not being played (#652)
  * Fix rendering of lock-picking messages
  * Fix some dialog lines with guru in G1 (#660)
  * Fix handling of fireDelay parameter in triggers (#636)
  * Add missing failureTarget trigger for CodeMaster trigger (#606)
  * Praying/Sitting npc's now standing up on player interaction (#615)

### New Contributors
* @houkama made their first contribution in https://github.com/Try/OpenGothic/pull/194
* @lmichaelis made their first contribution in https://github.com/Try/OpenGothic/pull/209
* @fhanrath made their first contribution in https://github.com/Try/OpenGothic/pull/219
* @matthiakl made their first contribution in https://github.com/Try/OpenGothic/pull/284
* @thokkat made their first contribution in https://github.com/Try/OpenGothic/pull/317
* @Reveares made their first contribution in https://github.com/Try/OpenGothic/pull/383
* @CReimer made their first contribution in https://github.com/Try/OpenGothic/pull/397
* @sjavora made their first contribution in https://github.com/Try/OpenGothic/pull/400
* @raphaelahrens made their first contribution in https://github.com/Try/OpenGothic/pull/415
* @yuriytaranov made their first contribution in https://github.com/Try/OpenGothic/pull/429
* @PoetaKodu made their first contribution in https://github.com/Try/OpenGothic/pull/441
* @striezel made their first contribution in https://github.com/Try/OpenGothic/pull/472
* @ThielHater made their first contribution in https://github.com/Try/OpenGothic/pull/515
* @KirillAlekseeenko made their first contribution in https://github.com/Try/OpenGothic/pull/572
* @hwnde made their first contribution in https://github.com/Try/OpenGothic/pull/583
* @Pinemarten made their first contribution in https://github.com/Try/OpenGothic/pull/650
* @Domarion made their first contribution in https://github.com/Try/OpenGothic/pull/672
* @LordKobra made their first contribution in https://github.com/Try/OpenGothic/pull/694

**Full Changelog**: https://github.com/Try/OpenGothic/compare/v0.48...opengothic-v1.0.2997

<hr /><em>This discussion was created from the release <a href='https://github.com/Try/OpenGothic/releases/tag/opengothic-v1.0.3010'>opengothic-v1.0.3010</a>.</em>

**Comment by TomislavTom** [comment]
Now make Archolos compatible 😀

**Reply by Try** [reply]
You can launch Archolos already, and it works until first cutscene (we missing some hooks there). However many pieces of mod already working:
![91](https://github.com/user-attachments/assets/f6a82ef7-ffda-4dde-b236-90b74fb37058)

**Reply by TomislavTom** [reply]
Cool! Hopefully we can play the entire game one day 😎

**Comment by Osorkonius** [comment]
So happy to see some new work! I was getting worried that OG was dropped :D
